### PR TITLE
[react] Add support for legacy context in class components in JSXElementConstructor

### PR DESCRIPTION
**Pull request by NMinhNguyen**
This is a follow-up to #65566 for class components referencing legacy context in the constructor.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65566

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react@latest/))
was missing the following properties:
1. unstable_act
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8bb235b618ebfb7de1230576ea14d3b70664bb18
</p>

File: types/react/test/tsx.tsx [hunk 1]
```diff
@@ -357,6 +357,22 @@ class LegacyContext extends React.Component {
...
+
+    render() {
+        // $ExpectType unknown
+        this.context;
+        return (this.context as any).foo;
+    }
+}
+
+<LegacyContextWithConstructor />;
+
```
Comment (eps1lon) [inline types/react/test/tsx.tsx:375]: Let's move this into `reactNodeTests` instead. It already has a group for "ReturnWithLegacyContext" so `RenderWithLegacyContext` fits right next to it.

Comment (NMinhNguyen) [reply types/react/test/tsx.tsx]: I’m guessing you actually meant `elementTypeTests`, so I moved the tests there.

**Review by eps1lon — APPROVED** [review]

**Comment by NMinhNguyen** [comment]
Ready to merge

**Pull request merged**